src/main/java/reactor/core/publisher/FluxWindowPredicate.java

```diff
@@ -266,66 +265,36 @@ public void onComplete() {
 		}
 
 		@Override
-		public long getCapacity() {
-			return prefetch;
-		}
-
-		@Override
-		public long getPending() {
-			return queue.size();
-		}
-
-		@Override
-		public boolean isStarted() {
-			return s != null && cancelled != 1 && !done;
-		}
-
-		@Override
-		public boolean isCancelled() {
-			return cancelled == 1;
-		}
-
-		@Override
-		public boolean isTerminated() {
-			return done;
-		}
-
-		@Override
-		public Throwable getError() {
-			return error;
-		}
-
-		@Override
-		public Iterator<?> downstreams() {
-			WindowGroupedFlux<T> g = window;
-			if (g == null) {
-				return Collections.emptyList()
-				                  .iterator();
+		public Object scan(Attr key) {
+			switch (key) {
+				case PARENT:
+					return s;
+				case CANCELLED:
+					return cancelled == 1;
+				case PREFETCH:
+					return prefetch;
+				case TERMINATED:
+					return done;
+				case BUFFERED:
+					return queue.size();
+				case ERROR:
+					return error;
+				case REQUESTED_FROM_DOWNSTREAM:
+					return requested;
 			}
-			return Collections.singletonList(g)
-			                  .iterator();
+			return InnerOperator.super.scan(key);
 		}
 
 		@Override
-		public long downstreamCount() {
-			return window == null ? 0L : 1L;
+		public Stream<? extends Scannable> inners() {
```

Comment:
:warning: `window` can and will be `null` at beginning. probably should protect against that with an empty stream!